src/test/java/org/prebid/server/bidder/openx/OpenxBidderTest.java

```diff
@@ -343,6 +343,70 @@ public void makeHttpRequestsShouldReturnResultWithExpectedFieldsSet() {
                                 .build());
     }
 
+    @Test
+    public void makeHttpRequestShouldReturnResultWithCustomBidFloorIfImpBidFloorIsZero() {
+        final BigDecimal customFloor = BigDecimal.valueOf(123);
+
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
+                        .bidfloor(BigDecimal.ZERO)
+                        .video(Video.builder().build())
+                        .ext(mapper.valueToTree(
+                                ExtPrebid.of(null,
+                                        ExtImpOpenx.builder()
+                                                .customFloor(customFloor).build())))
+                        .build()))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = openxBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getValue()).hasSize(1)
+                .extracting(httpRequest -> mapper.readValue(httpRequest.getBody(), BidRequest.class))
+                .containsExactly(BidRequest.builder()
+                        .imp(singletonList(Imp.builder()
+                                .bidfloor(customFloor)
+                                .video(Video.builder().build())
+                                .build()))
+                        .ext(jacksonMapper.fillExtension(
+                                ExtRequest.empty(), OpenxRequestExt.of(null, null, "hb_pbs_1.0.0")))
+                        .build());
```

Comment:
Just floor check is enough

Reply:
Actually just floor check is not enough because of this explicit `.ext` fulfilment: https://github.com/prebid/prebid-server-java/blob/master/src/main/java/org/prebid/server/bidder/openx/OpenxBidder.java#L176-L181